### PR TITLE
Jade parser improvements

### PR DESCRIPTION
**Pull request by pirxpilot**
- use standard jsxgettext test infrastructure
- add support for *pipeless* multiline text.
- add support for L10n comments

Comments, suggestions?

### CODE REVIEW

File: lib/parsers/jade.js [hunk 1]
```diff
@@ -34,24 +34,42 @@ function parseJade(str, options) {
...
+  var buf = [], lineN, lineAdjust = -1;
+
+  function append(text) {
+    /* jshint -W040 */
+    var ctx = this;
+    if (ctx && ctx.bump) {
+      lineN += 1;
+    }
+    if (text.length) {
+      buf[lineN] = [buf[lineN], text, ';'].join('');
```
Comment (BYK) [inline lib/parsers/jade.js]: It's probably better for you to just string concat here.

File: lib/parsers/jade.js [hunk 2]
```diff
@@ -34,24 +34,42 @@ function parseJade(str, options) {
...
+        break;
+      case 'pipeless-text':
+        token.val
+          .map(extractGettext)
+          .forEach(append, { bump: true });
+        lineAdjust += token.val.length;
+        break;
+      case 'comment':
+        if (/^\s*L10n:/.test(token.val)) {
+          append(['/*', token.val, '*/'].join(''));
```
Comment (BYK) [inline lib/parsers/jade.js]: Same as above, just use normal string concat?

File: test/tests/jade.js [hunk 3]
```diff
@@ -6,35 +6,21 @@ var path = require('path');
...
-    assert.ok(result.indexOf('msgid ' + msg) > -1, msg + ' is found');
-  }
-
-  // check that files with leading hash parse
-  var inputFilename = path.join(__dirname, '..', 'inputs', 'second_attribute.jade');
+  var inputFilename = path.join(__dirname, '..', 'inputs', 'example.jade');
   fs.readFile(inputFilename, "utf8", function (err, source) {
-
     var opts = {keyword: ['gettext', '_']},
-        sources = {'inputs/second_attribute.jade': source},
```
Comment (BYK) [inline test/tests/jade.js]: Should we remove this file altogether?

**Comment by BYK** [comment]
>Comments, suggestions?

You rock? :+1: :)

File: lib/parsers/jade.js (hunk 1 above)
Comment (pirxpilot) [inline lib/parsers/jade.js]: Better for me? :-)

    buf[lineN] = (buf[lineN] || '') + text + ';';

If you insist I can change it: I just think version with `join` seems cleaner.

File: lib/parsers/jade.js (hunk 2 above)
Comment (pirxpilot) [inline lib/parsers/jade.js]: OK

File: test/tests/jade.js (hunk 3 above)
Comment (pirxpilot) [inline test/tests/jade.js]: I was actually thinking about it. But there is a second test that is using `second_attribute.jade` - and I was not entirely comfortable just removing it.
Any idea what exactly is tested here - seems to be related to #40
ping @mweibel

File: lib/parsers/jade.js (hunk 1 above)
Comment (BYK) [inline lib/parsers/jade.js]: Not insisting :)

File: test/tests/jade.js (hunk 3 above)
Comment (BYK) [inline test/tests/jade.js]: We can do it in a follow-up commit then. Thanks for looking into this.

**Pull request merged**